TheMET/TheMET/Services/FeaturedArtService.swift

```diff
@@ -0,0 +1,148 @@
+        case .loading:
+            return
+        case .loaded(let art):
+            guard let storedFeaturedArtData = self.readFeaturedArtDataFromStore(),
+                  art.objectID == storedFeaturedArtData.artId,
+                  self.isFeaturedArtDateActual(date: storedFeaturedArtData.date) else {
+                self.forceUpdateFeaturedArt()
+                return
+            }
+            return
+        }
+    }
+    
+    static let artIDKey: String = "artID"
+    static let dateKey: String = "date"
+    
+    private func storeFeaturedArtData(artId: ArtID, date: Date) {
+        UserDefaults.standard.set(artId, forKey: FeaturedArtService.artIDKey)
+        let dateString = self.getDateString(date: date)
+        UserDefaults.standard.set(dateString, forKey: FeaturedArtService.dateKey)
+    }
+    
+    private func readFeaturedArtDataFromStore() -> (artId: ArtID, date: Date)? {
+        let id = UserDefaults.standard.integer(forKey: FeaturedArtService.artIDKey)
+        guard id != 0 else {
+            return nil
+        }
+        guard let dateString = UserDefaults.standard.string(forKey: FeaturedArtService.dateKey),
+              let date = self.getDate(dateString: dateString) else {
+            return nil
+        }
+        return (id, date)
+    }
+    
+    private func updateFeaturedArt() {
+        guard let storedData = self.readFeaturedArtDataFromStore(),
+        self.isFeaturedArtDateActual(date: storedData.date),
+        let artFileManager = self.artFileManager else {
+            self.forceUpdateFeaturedArt()
+            return
+        }
+        artFileManager.readArt(id: storedData.artId, completion: { [weak self] art in
+            guard let art = art else {
+                self?.forceUpdateFeaturedArt()
+                return
+            }
+            self?.featuredArt = .loaded(art)
+        })
+    }
+    
+    private func getDateString(date: Date) -> String {
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "dd/MM/yyyy"
+        return dateFormatter.string(from: date)
+    }
+    
+    private func getDate(dateString: String) -> Date? {
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "dd/MM/yyyy"
+        return dateFormatter.date(from: dateString)
+    }
+    
+    private func isFeaturedArtDateActual(date: Date) -> Bool {
+        let currentDateString = self.getDateString(date: Date.now)
+        let dateString = self.getDateString(date: date)
+        return currentDateString == dateString
+    }
+    
+    private func forceUpdateFeaturedArt() {
+        guard !self.isFeaturedArtLoading else {
+            return
+        }
+        self.isFeaturedArtLoading = true
+        self.metAPI.objects { [weak self] objecResponce in
+            guard let objecResponce = objecResponce,
+            let randomId = objecResponce.objectIDs.randomElement() else {
+                self?.featuredArt = .failedToLoad
```

Comment:
When loading fails, the `isFeaturedArtLoading` must also be switched back to `false`. The same at line 136

Reply:
fixed